relay-server/src/actors/processor.rs

```diff
@@ -4437,4 +4451,96 @@ mod tests {
 
         assert_eq!(get_sampling_match(res).sample_rate(), 0.4);
     }
+
+    #[tokio::test]
+    async fn test_profile_id_transfered() {
```

Comment:
does this have to be `tokio::test` and `async`?

Reply:
👍 It can be sync if I use dummy `Addr`s instead of `mock_service`. Changed here: https://github.com/getsentry/relay/pull/2733/commits/e68e1badb5f48193c2c6aed46a56832d9a157155